### PR TITLE
doc: add tidb rollup syntax

### PR DESCRIPTION
**Pull request by qiancai**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/15053
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/group-by-modifier.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+In the preceding example, the order of columns in the grouping list is `[year, month]`, and the dimension groups generated by the ROLLUP syntax are `{year, month}`, `{year}`, and `{}`. For the dimension group `{year, month}`, both `year` and `month` are the columns required, so TiDB fills the bit positions for them with 1 and 1 correspndingly. This forms a UINT64 of `11...0`, which is 3 in decimal. Therefore, the projection expression is `[test.bank.profit, Column#6, Column#7, 3->gid]` (where `column#6` corresponds to `year`, and `column#7` corresponds to `month`).
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:190]:
```suggestion
In the preceding example, the order of columns in the grouping list is `[year, month]`, and the dimension groups generated by the ROLLUP syntax are `{year, month}`, `{year}`, and `{}`. For the dimension group `{year, month}`, both `year` and `month` are the columns required, so TiDB fills the bit positions for them with 1 and 1 correspondingly. This forms a UINT64 of `11...0`, which is 3 in decimal. Therefore, the projection expression is `[test.bank.profit, Column#6, Column#7, 3->gid]` (where `column#6` corresponds to `year`, and `column#7` corresponds to `month`).
```

File: functions-and-operators/group-by-modifier.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+| 2001 | NULL  | 22.399999618530273 |        0 |         1 |
+| 2000 | NULL  |  41.90000057220459 |        0 |         1 |
+| NULL | NULL  |  64.30000019073486 |        1 |         1 |
++------+-------+--------------------+----------+-----------+
+3 rows in set (0.023 sec)
+```
+
+## How to interpret the ROLLUP execution plan
+
+To meet the requirements of multidimensional grouping, multidimensional data aggregation uses the `Expand` operator to replicate data. Each replica corresponds to a group at a specific dimension. With the data shuffling capability of MPP, the `Expand` operator can rapidly reorganize and calculates a large volume of data between multiple TiFlash nodes, fully utilizing the computational power of each node.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:161]:
```suggestion
To meet the requirements of multidimensional grouping, multidimensional data aggregation uses the `Expand` operator to replicate data. Each replica corresponds to a group at a specific dimension. With the data shuffling capability of MPP, the `Expand` operator can rapidly reorganize and calculate a large volume of data between multiple TiFlash nodes, fully utilizing the computational power of each node.
```

File: functions-and-operators/group-by-modifier.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+The implementation of the `Expand` operator is similar to that of the `Projection` operator. The difference is that `Expand` is a multi-level `Projection`, which contains multiple levels of projection operation expressions. For each row of the raw data, the `Projection` operator generates only one row in results, whereas the `Expand` operator generates multiple rows in results (the number of rows equals to the number of levels in projection operation expressions).
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:163]:
```suggestion
The implementation of the `Expand` operator is similar to that of the `Projection` operator. The difference is that `Expand` is a multi-level `Projection`, which contains multiple levels of projection operation expressions. For each row of the raw data, the `Projection` operator generates only one row in results, whereas the `Expand` operator generates multiple rows in results (the number of rows equals the number of levels in projection operation expressions).
```

File: functions-and-operators/group-by-modifier.md [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+10 rows in set (0.05 sec)
+```
+
+In this example execution plan, you can view the multiple-level expression of the `Expand` operator in the `operator info` column of the `Expand_20` row. It consists of 2-dimensional expressions, and you can view the schema information of the `Expand` operator at the end of the row, which is `schema: [test.bank.profit, Column#6, Column#7, gid]`.
+
+In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, reflecting the relationship between the current data replica and the grouping dimension. In the most cases, the `Expand` operator uses a bitmask operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the group of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:188]:
```suggestion
In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, reflecting the relationship between the current data replica and the grouping dimension. In most cases, the `Expand` operator uses a bitmask operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the group of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```

File: functions-and-operators/group-by-modifier.md [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+In the preceding example, the order of columns in the grouping list is `[year, month]`, and the dimension groups generated by the ROLLUP syntax are `{year, month}`, `{year}`, and `{}`. For the dimension group `{year, month}`, both `year` and `month` are the columns required, so TiDB fills the bit positions for them with 1 and 1 correspndingly. This forms a UINT64 of `11...0`, which is 3 in decimal. Therefore, the projection expression is `[test.bank.profit, Column#6, Column#7, 3->gid]` (where `column#6` corresponds to `year`, and `column#7` corresponds to `month`).
+
+The following is a example row of the raw data:
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:192]:
```suggestion
The following is an example row of the raw data:
```

File: explain-aggregation.md [hunk 6]
```diff
@@ -170,3 +170,36 @@ Query OK, 0 rows affected (0.28 sec)
 +------------------------------+---------+-----------+----------------------------+----------------------------------------------------------------------------------------------------+
 5 rows in set (0.00 sec)
 ```
+
+## Multidimensional data aggregation with ROLLUP
+
+Starting from v7.4.0, the `GROUP BY` clause of TiDB supports the `WITH ROLLUP` modifier.
+
+In the `GROUP BY` clause, you can specify one or more columns as a group list, and then append the `WITH ROLLUP` modifier to the list. After that, TiDB will conduct multidimensional descending grouping based on the columns in the group list and provide you with summary results for each group in the output.
```
Comment (qiancai) [inline explain-aggregation.md:178]:
```suggestion
In the `GROUP BY` clause, you can specify one or more columns as a group list and append the `WITH ROLLUP` modifier after the list. Then, TiDB will conduct multidimensional descending grouping based on the columns in the group list and provide you with summary results for each group in the output.
```

File: explain-aggregation.md [hunk 7]
```diff
@@ -170,3 +170,36 @@ Query OK, 0 rows affected (0.28 sec)
...
+|                 └─TableFullScan_17     | 3.00    | mpp[tiflash] | table:bank    | keep order:false, stats:pseudo                                                                                                                                                                                                       |
++----------------------------------------+---------+--------------+---------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+10 rows in set (0.05 sec)
+```
+
+According to the `GROUP BY year, month WITH ROLLUP` syntax in the preceding statement, the SQL aggregation for this statement can be calculated and concatenated in three groups: `{year, month}`, `{year}`, and `{}` respectively.
```
Comment (qiancai) [inline explain-aggregation.md:203]:
```suggestion
According to the `GROUP BY year, month WITH ROLLUP` syntax in the preceding statement, the SQL aggregation results for this statement can be calculated and concatenated in three groups: `{year, month}`, `{year}`, and `{}` respectively.
```

File: functions-and-operators/group-by-modifier.md [hunk 8]
```diff
@@ -0,0 +1,216 @@
+---
+title: GROUP BY Modifiers
+summary: Learn how to use TiDB GROUP BY modifiers.
+---
+
+# GROUP BY Modifiers
+
+Starting from v7.4.0, the `GROUP BY` clause of TiDB supports the `WITH ROLLUP` modifier.
+
+In the `GROUP BY` clause, you can specify one or more columns as a group list, and then append the `WITH ROLLUP` modifier to the list. After that, TiDB will conduct multidimensional descending grouping based on the columns in the group list and provide you with summary results for each group in the output.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:10]:
```suggestion
In the `GROUP BY` clause, you can specify one or more columns as a group list and append the `WITH ROLLUP` modifier after the list. Then, TiDB will conduct multidimensional descending grouping based on the columns in the group list and provide you with summary results for each group in the output.
```

File: functions-and-operators/group-by-modifier.md [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+
+- Grouping method:
+
+    - The first grouping dimension includes all columns in the group list.
+    - Subsequent grouping dimensions start from the right end of the grouping list and decrease one element at a time to form new groups.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:15]:
```suggestion
    - Subsequent grouping dimensions start from the right end of the grouping list and decrease one column at a time to form new groups.
```

File: functions-and-operators/group-by-modifier.md [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+- Aggregation summaries: on each dimension, the query performs aggregation operations, and then aggregates the results of this dimension with the results of all previous dimensions. This means you can get aggregated data at different dimensions, from detailed to overall.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:17]:
```suggestion
- Aggregation summaries: for each dimension, the query performs aggregation operations, and then aggregates the results of this dimension with the results of all previous dimensions. This means that you can get aggregated data at different dimensions, from detailed to overall.
```

File: functions-and-operators/group-by-modifier.md [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
++------+--------------------+
+| year | profit             |
++------+--------------------+
+| 2001 | 22.399999618530273 |
+| 2000 |  41.90000057220459 |
++------+--------------------+
+2 rows in set (0.15 sec)
+```
+
+In addition to yearly profits, bank reports usually also need to include the overall profit for all years and monthly profits for in-depth or more detailed profit analysis. Before v7.4.0, you have to use different `GROUP BY` clauses in multiple queries and join the results using UNION to obtain aggregated summaries. Starting from v7.4.0, you can simply achieve the desired results by appending the `WITH ROLLUP` modifier to the `GROUP BY` clause in a single query.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:74]:
```suggestion
In addition to yearly profits, bank reports usually also need to include the overall profit for all years and monthly profits for in-depth or more detailed profit analysis. Before v7.4.0, you have to use different `GROUP BY` clauses in multiple queries and join the results using UNION to obtain aggregated summaries. Starting from v7.4.0, you can simply achieve the desired results in a single query by appending the `WITH ROLLUP` modifier to the `GROUP BY` clause.
```

File: functions-and-operators/group-by-modifier.md [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+
+Specifically:
+
+- The `profit` value in the first row comes from the 2-dimensional group `{year, month}`, representing the aggregation result for the fine-grained `{2000, "Jan"}` group.
+- The `profit` value in the second row comes from the 1-dimensional group `{year}`, representing the aggregation result for the mid-level `{2001}` group.
+- The `profit` value in the last row comes from the 0-dimensional grouping `{}`, representing the overall aggregation result.
+
+`NULL` values in the `WITH ROLLUP` results are generated before the Aggregate operator is applied. Therefore, you can use `NULL` values in `SELECT`, `HAVING`, and `ORDER BY` clauses to further filter the aggregated results.
+
+For example, you can use `NULL` in the `HAVING` clause to filter and view the aggregated results for the 2-dimensional groups only:
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:101]:
```suggestion
For example, you can use `NULL` in the `HAVING` clause to filter and view the aggregated results of 2-dimensional groups only:
```

File: functions-and-operators/group-by-modifier.md [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+| 2000 | Mar   | 31.600000381469727 |
+| 2000 | Jan   | 10.300000190734863 |
+| 2001 | Feb   | 22.399999618530273 |
++------+-------+--------------------+
+3 rows in set (0.02 sec)
+```
+
+Note that if a column in the `GROUP BY` list contains native `NULL` values, the aggregation results of `WITH ROLLUP` might mislead the query results. To address this issue, you can use the `GROUPING()` function to distinguish native `NULL` values from `NULL` values generated by `WITH ROLLUP`. This function takes a grouping expression as a parameter and returns `0` or `1` to indicate whether the grouping expression is aggregated in the current result. `1` represents aggregated, and `0` represents not aggregated.
+
+The following is an example of how to use the `GROUPING()` function:
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:117]:
```suggestion
The following example shows how to use the `GROUPING()` function:
```

File: functions-and-operators/group-by-modifier.md [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+| 2001 | NULL  | 22.399999618530273 |        0 |         1 |
+| 2000 | Mar   | 31.600000381469727 |        0 |         0 |
+| 2000 | Jan   | 10.300000190734863 |        0 |         0 |
+| 2000 | NULL  |  41.90000057220459 |        0 |         1 |
+| NULL | NULL  |  64.30000019073486 |        1 |         1 |
++------+-------+--------------------+----------+-----------+
+6 rows in set (0.028 sec)
+```
+
+From this output, you can determine the aggregation dimension of a row directly from the results of `grp_year` and `grp_month`, which prevents interference from native `NULL` values in the `year` and `month` grouping expressions.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:134]:
```suggestion
From this output, you can get the aggregation dimension of a row directly from the results of `grp_year` and `grp_month`, which prevents interference from native `NULL` values in the `year` and `month` grouping expressions.
```

File: functions-and-operators/group-by-modifier.md [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+10 rows in set (0.05 sec)
+```
+
+In this example execution plan, you can view the multiple-level expression of the `Expand` operator in the `operator info` column of the `Expand_20` row. It consists of 2-dimensional expressions, and you can view the schema information of the `Expand` operator at the end of the row, which is `schema: [test.bank.profit, Column#6, Column#7, gid]`.
+
+In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, reflecting the relationship between the current data replica and the grouping dimension. In most cases, the `Expand` operator uses a bitmask operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the group of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:188]:
```suggestion
In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, and the value reflects the relationship between the current data replica and the grouping dimension. In most cases, the `Expand` operator uses a bitmask operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the group of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```

File: functions-and-operators/group-by-modifier.md [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+
+In the preceding example, the order of columns in the grouping list is `[year, month]`, and the dimension groups generated by the ROLLUP syntax are `{year, month}`, `{year}`, and `{}`. For the dimension group `{year, month}`, both `year` and `month` are the columns required, so TiDB fills the bit positions for them with 1 and 1 correspondingly. This forms a UINT64 of `11...0`, which is 3 in decimal. Therefore, the projection expression is `[test.bank.profit, Column#6, Column#7, 3->gid]` (where `column#6` corresponds to `year`, and `column#7` corresponds to `month`).
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:190]:
```suggestion
In the preceding example, the order of columns in the grouping list is `[year, month]`, and the dimension groups generated by the ROLLUP syntax are `{year, month}`, `{year}`, and `{}`. For the dimension group `{year, month}`, both `year` and `month` are required columns, so TiDB fills the bit positions for them with 1 and 1 correspondingly. This forms a UINT64 of `11...0`, which is 3 in decimal. Therefore, the projection expression is `[test.bank.profit, Column#6, Column#7, 3->gid]` (where `column#6` corresponds to `year`, and `column#7` corresponds to `month`).
```

File: functions-and-operators/group-by-modifier.md [hunk 17]
```diff
@@ -0,0 +1,216 @@
...
+| 2001 | NULL  | 22.399999618530273 |
+| 2000 | Mar   | 31.600000381469727 |
+| 2000 | Jan   | 10.300000190734863 |
+| 2000 | NULL  |  41.90000057220459 |
+| NULL | NULL  |  64.30000019073486 |
++------+-------+--------------------+
+6 rows in set (0.025 sec)
+```
+
+The preceding results include aggregated data at different dimensions: by year, by month, and overall. In the results, a row without `NULL` values indicates that the `profit` in that row is calculated by grouping both year and month. A row with a `NULL` value in the `month` column indicates that `profit` in that row is calculated by aggregating months, while a row with a `NULL` value in the `year` column indicates that `profit` in that row is calculated by aggregating years.
```
Comment (AilinKid) [inline functions-and-operators/group-by-modifier.md:91]:
```suggestion
The preceding results include aggregated data at different dimensions: by year, by year and month, and overall. In the results, a row without `NULL` values indicates that the `profit` in that row is calculated by grouping both year and month. A row with a `NULL` value in the `month` column indicates that `profit` in that row is calculated by aggregating months, while a row with a `NULL` value in the `year` column indicates that `profit` in that row is calculated by aggregating years.
```

File: functions-and-operators/group-by-modifier.md [hunk 18]
```diff
@@ -0,0 +1,216 @@
...
+
+Specifically:
+
+- The `profit` value in the first row comes from the 2-dimensional group `{year, month}`, representing the aggregation result for the fine-grained `{2000, "Jan"}` group.
+- The `profit` value in the second row comes from the 1-dimensional group `{year}`, representing the aggregation result for the mid-level `{2001}` group.
+- The `profit` value in the last row comes from the 0-dimensional grouping `{}`, representing the overall aggregation result.
+
+`NULL` values in the `WITH ROLLUP` results are generated before the Aggregate operator is applied. Therefore, you can use `NULL` values in `SELECT`, `HAVING`, and `ORDER BY` clauses to further filter the aggregated results.
```
Comment (AilinKid) [inline functions-and-operators/group-by-modifier.md:99]:
```suggestion
`NULL` values in the `WITH ROLLUP` results are generated just before the Aggregate operator is applied. Therefore, you can use `NULL` values in `SELECT`, `HAVING`, and `ORDER BY` clauses to further filter the aggregated results.
```

File: functions-and-operators/group-by-modifier.md [hunk 19]
```diff
@@ -0,0 +1,216 @@
...
+| 2001 | NULL  | 22.399999618530273 |        0 |         1 |
+| 2000 | Mar   | 31.600000381469727 |        0 |         0 |
+| 2000 | Jan   | 10.300000190734863 |        0 |         0 |
+| 2000 | NULL  |  41.90000057220459 |        0 |         1 |
+| NULL | NULL  |  64.30000019073486 |        1 |         1 |
++------+-------+--------------------+----------+-----------+
+6 rows in set (0.028 sec)
+```
+
+From this output, you can get the aggregation dimension of a row directly from the results of `grp_year` and `grp_month`, which prevents interference from native `NULL` values in the `year` and `month` grouping expressions.
```
Comment (AilinKid) [inline functions-and-operators/group-by-modifier.md:134]:
```suggestion
From this output, you can get an understanding of the aggregation dimension of a row directly from the results of `grp_year` and `grp_month`, which prevents interference from native `NULL` values in the `year` and `month` grouping expressions.
```

File: functions-and-operators/group-by-modifier.md [hunk 20]
```diff
@@ -0,0 +1,216 @@
...
+10 rows in set (0.05 sec)
+```
+
+In this example execution plan, you can view the multiple-level expression of the `Expand` operator in the `operator info` column of the `Expand_20` row. It consists of 2-dimensional expressions, and you can view the schema information of the `Expand` operator at the end of the row, which is `schema: [test.bank.profit, Column#6, Column#7, gid]`.
+
+In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, and the value reflects the relationship between the current data replica and the grouping dimension. In most cases, the `Expand` operator uses a bitmask operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the group of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```
Comment (AilinKid) [inline functions-and-operators/group-by-modifier.md:188]:
```suggestion
In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, and the value reflects the relationship between the current data replica and the `grouping set`. In most cases, the `Expand` operator uses a Bit-And operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the group set of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

File: functions-and-operators/group-by-modifier.md [hunk 21]
```diff
@@ -0,0 +1,216 @@
...
+10 rows in set (0.05 sec)
+```
+
+In this example execution plan, you can view the multiple-level expression of the `Expand` operator in the `operator info` column of the `Expand_20` row. It consists of 2-dimensional expressions, and you can view the schema information of the `Expand` operator at the end of the row, which is `schema: [test.bank.profit, Column#6, Column#7, gid]`.
+
+In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, and the value reflects the relationship between the current data replica and the `grouping set`. In most cases, the `Expand` operator uses a Bit-And operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the group set of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:188]:
```suggestion
In the schema information of the `Expand` operator, `GID` is generated as an additional column. Its value is calculated by the `Expand` operator based on the grouping logic of different dimensions, and the value reflects the relationship between the current data replica and the `grouping set`. In most cases, the `Expand` operator uses a Bit-And operation, which can represent 63 combinations of grouping items for ROLLUP, corresponding to 64 dimensions of grouping. In this mode, TiDB generates the `GID` value depending on whether the `grouping set` of the required dimension contains grouping expressions when the current data replica is replicated, and it fills a 64-bit UINT64 value in the order of columns to be grouped.
```

File: functions-and-operators/group-by-modifier.md [hunk 22]
```diff
@@ -0,0 +1,216 @@
...
++------------+------+-------+-----+
+| 10.3000000 | 2000 | Jan   |  3  |
++------------+------+-------+-----+
+| 10.3000000 | 2000 | NULL  |  1  |
++------------+------+-------+-----+
+| 10.3000000 | NULL | NULL  |  0  |
++------------+------+-------+-----+
+```
+
+Note that the `SELECT` clause in the query uses the `GROUPING` function. When the `GROUPING` function is used in the `SELECT`, `HAVING`, or `ORDER BY` clauses, TiDB rewrites it during the logical optimization phase, transforms the relationship between the `GROUPING` function and the `GROUP BY` items into a `GID` related to the logic of dimension group (or known as `grouping set`), and fills this `GID` as metadata into the new `GROUPING` function.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:216]:
```suggestion
Note that the `SELECT` clause in the query uses the `GROUPING` function. When the `GROUPING` function is used in the `SELECT`, `HAVING`, or `ORDER BY` clauses, TiDB rewrites it during the logical optimization phase, transforms the relationship between the `GROUPING` function and the `GROUP BY` items into a `GID` related to the logic of dimension group (also known as `grouping set`), and fills this `GID` as metadata into the new `GROUPING` function.
```

File: functions-and-operators/group-by-modifier.md [hunk 23]
```diff
@@ -0,0 +1,216 @@
...
+| 2001 | NULL  | 22.399999618530273 |
+| 2000 | Mar   | 31.600000381469727 |
+| 2000 | Jan   | 10.300000190734863 |
+| 2000 | NULL  |  41.90000057220459 |
+| NULL | NULL  |  64.30000019073486 |
++------+-------+--------------------+
+6 rows in set (0.025 sec)
+```
+
+The preceding results include aggregated data at different dimensions: by year, by both year and month, and overall. In the results, a row without `NULL` values indicates that the `profit` in that row is calculated by grouping both year and month. A row with a `NULL` value in the `month` column indicates that `profit` in that row is calculated by aggregating months, while a row with a `NULL` value in the `year` column indicates that `profit` in that row is calculated by aggregating years.
```
Comment (qiancai) [inline functions-and-operators/group-by-modifier.md:91]:
```suggestion
The preceding results include aggregated data at different dimensions: by both year and month, by year, and overall. In the results, a row without `NULL` values indicates that the `profit` in that row is calculated by grouping both year and month. A row with a `NULL` value in the `month` column indicates that `profit` in that row is calculated by aggregating months, while a row with a `NULL` value in the `year` column indicates that `profit` in that row is calculated by aggregating years.
```

**Review by AilinKid — APPROVED** [review]

File: functions-and-operators/group-by-modifier.md [hunk 24]
```diff
@@ -0,0 +1,216 @@
...
+| NULL | NULL  |  64.30000019073486 |        1 |         1 |
++------+-------+--------------------+----------+-----------+
+3 rows in set (0.023 sec)
+```
+
+## How to interpret the ROLLUP execution plan
+
+To meet the requirements of multidimensional grouping, multidimensional data aggregation uses the `Expand` operator to replicate data. Each replica corresponds to a group at a specific dimension. With the data shuffling capability of MPP, the `Expand` operator can rapidly reorganize and calculate a large volume of data between multiple TiFlash nodes, fully utilizing the computational power of each node.
+
+The implementation of the `Expand` operator is similar to that of the `Projection` operator. The difference is that `Expand` is a multi-level `Projection`, which contains multiple levels of projection operation expressions. For each row of the raw data, the `Projection` operator generates only one row in results, whereas the `Expand` operator generates multiple rows in results (the number of rows equals the number of levels in projection operation expressions).
```
Comment (hfxsd) [inline functions-and-operators/group-by-modifier.md:163]:
```suggestion
The implementation of the `Expand` operator is similar to that of the `Projection` operator. The difference is that `Expand` is a multi-level `Projection`, which contains multiple levels of projection operation expressions. For each row of the raw data, the `Projection` operator generates only one row in results, whereas the `Expand` operator generates multiple rows in results (the number of rows equals to the number of levels in projection operation expressions).
```

**Review by hfxsd — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Pull request merged**